core/src/main/java/org/stellar/anchor/util/NetUtil.java

```diff
@@ -17,7 +17,14 @@ public static String fetch(String url) throws IOException {
     Request request = OkHttpUtil.buildGetRequest(url);
     Response response = getCall(request).execute();
 
-    if (response.body() == null) return "";
+    // Check if response was successful (status code 200)
+    if (!response.isSuccessful()) {
+      throw new IOException(String.format("Unable to fetch data from %s", url));
```

Comment:
I am not sure how this would fix the SSRF issue. But if you think it is the fix, please feel free to merge.

Reply:
just to be consistent. returning "" will be more confusing i think. if the response is empty, we want to throw an exception as well.

Reply:
also, this code is within a method that is responsible for handling I/O operations, i feel appropriate to throw an IOException. This would signal to callers that an I/O error occurred, which could include an empty response body from a network operation.  that is throwing, IOException, is typically used to signal an error related to Input/Output operations, such as reading or writing to files or network resources. In this context, we are checking if the response body is empty, thus an appropriate exception to throw in the context of reading data from a network resource.